c7n/filters/kms.py

```diff
@@ -42,13 +59,13 @@ def get_related_ids(self, resources):
         return normalized_ids
 
     def process(self, resources, event=None):
-        client = local_session(self.manager.session_factory).client('kms')
         related = self.get_related(resources)
         for r in related.values():
             try:
-                alias_info = self.manager.retry(client.list_aliases, KeyId=r.get('KeyId'))
-            except ClientError as e:
-                self.log.warning(e)
-                continue
-            r['c7n:AliasName'] = alias_info.get('Aliases')[0].get('AliasName', '')
+                # `AliasNames` is set when we fetch keys, but only for keys
+                # which have aliases defined. Fall back to an empty string
+                # to avoid lookup errors in filters.
+                r['c7n:AliasName'] = r['AliasNames'][0]
```

Comment:
how about `r.get('AliasNames', ('',))[0]`

Reply:
Good call, that's friendlier :+1: